File: CrackingTheCodingInterview-Android/src/commonMain/kotlin/Chapter1/CheckPermutationSolver.kt
package Chapter1

/**
 * CheckPermutationSolver
 *
 * @constructor Create empty CheckPermutationSolver
 */
public class CheckPermutationSolver {
    init {}

    /**
     * Checks whether two strings a permutations of one another. Time complexity is O(n log n).
     *
     * @param input The first string value.
     * @param input2 The second string value.
     * @return A boolean that indicates whether the strings are permutation of one another.
     */
    public fun checkPermutationWithSort(input: String, input2: String) : Boolean {
        if (input == input2) { return false }
        return input.toCharArray().sorted() == input2.toCharArray().sorted()
    }
}

File: CrackingTheCodingInterview-iOS/CrackingTheCodingInterview/Chapter 1/CheckPermutationSolver.swift
//
//  CheckPermutation.swift
//  CrackingTheCodingInterview
//
//  Created by Kurt Jacobs on 2022/12/16.
//

import Foundation

public class CheckPermutationSolver {
    public init() {}

    /// Checks whether two strings a permutations of one another. Time complexity is O(n log n) as indicated on https://developer.apple.com/documentation/swift/string/unicodescalarview/sorted()
    /// - Parameters:
    ///   - input: The first string value.
    ///   - input2: The second string value.
    /// - Returns: A boolean that indicates whether or not the strings are permutation of one another.
    @discardableResult
    public func checkPermutationWithSort(_ input: String, _ input2: String) -> Bool {
        guard input != input2 else { return false}
        return input.sorted() == input2.sorted()
    }

}
